### PR TITLE
CI: update rust toolchains

### PR DESCRIPTION
**Pull request by dannywillems**
See https://github.com/o1-labs/proof-systems/pull/1650#discussion_r1473679152

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
   run_checks:
     strategy:
       matrix:
-        rust_toolchain_version: ["1.71", "1.72", "1.73", "1.74"]
+        rust_toolchain_version: ["1.71", "1.72", "1.73", "1.74", "1.75", "1.76", "beta"]
```
Comment (volhovm) [inline .github/workflows/rust.yml:18]: Are you sure about `beta` though?..

**Review by volhovm — APPROVED** [review]
New clippy -- new life! :tada:

File: .github/workflows/rust.yml (hunk 1 above)
Comment (volhovm) [reply .github/workflows/rust.yml:18]: + do we have to support 1.71 still? CI might not like building 75% more versions than it does now.

**Comment by dannywillems** [comment]
Oh no, the doc does not build...
```
error: internal compiler error: compiler/rustc_middle/src/ty/generic_args.rs:184:43: expected a const, but found another kind
```

**Comment by dannywillems** [comment]
Gonna close it as old.

**Pull request closed without merging**